extensions/avro-extensions/src/main/java/io/druid/data/input/avro/AvroValueInputFormat.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.druid.data.input.avro;
+
+import org.apache.avro.Schema;
+import org.apache.avro.mapred.AvroValue;
+import org.apache.avro.mapreduce.AvroJob;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+public class AvroValueInputFormat<T> extends FileInputFormat<NullWritable, AvroValue<T>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AvroValueInputFormat.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RecordReader<NullWritable, AvroValue<T>> createRecordReader(
+            InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+        Schema readerSchema = AvroJob.getInputKeySchema(context.getConfiguration());
```

Comment:
I agree with @will-lauer that the schema should be specified as a path to a file on the hadoop FS.